### ISSUE TITLE
[BUG] Eww maxes single core of my CPU

### ISSUE THREAD

**Issue by awsomearvinder** [elkowar/eww#564]
### Checklist before submitting an issue

- [X] I have searched through the existing [closed and open issues](https://github.com/elkowar/eww/issues?q=is%3Aissue) for eww and made sure this is not a duplicate
- [X] I have specifically verified that this bug is not a common [user error](https://github.com/elkowar/eww/issues?q=is%3Aissue+label%3Ano-actual-bug+is%3Aclosed)
- [X] I am providing as much relevant information as I am able to in this bug report (Minimal config to reproduce the issue for example, if applicable)

### Description of the bug

I've been using eww for a little while now, and I really like it! However, an issue occurs when eww is open too long where it freezes on me and stops working. I'm not sure if this is related to #384 because this only affects a single thread, which means for me it tops out at 8% or so. After generating a flamegraph - it seems the culprit was `eww::state::one_to_n_elements_map`. After adding a debug statement in there, it seems like the length of `children` seems to keep growing linearly forever.
 ![flamegraph](https://user-images.githubusercontent.com/45527519/189551614-8b655162-02ef-466f-9a31-1e5e8b579804.svg)

### Reproducing the issue

My configs are at: https://github.com/awsomearvinder/nixhomeconfiguration/tree/f5999533db9b4a55b090284a725643f29e4c2f01/bender/dotfiles/eww. My configs are fairly minimal as is, but if I had to guess I'd imagine that the issue is just the low defpoll timer? I haven't tested that quite yet.

### Expected behaviour

CPU usage to remain constant

**Comment by awsomearvinder** [comment]
minimal example:
```eww
(defwindow example
           :monitor 0
           :geometry (geometry  :x 0
                                :y 0
                                :height "20px"
                                :width "100%"
                                :anchor "top center")
           :stacking "fg"
           :exclusive true
   (box
     (for entry in a
           (button "stuff"))))


(defpoll a :interval "1ms"
  'echo [3]')
  ```
  It seems like the for loop needs to take a defpoll for the issue to occur.

**Comment by elkowar** [comment]
This should now be fixed by 0cccd9d
